File: MyBooks/MyBooks.Service/Database/KorisnikZnacka.cs
using System;
using System.Collections.Generic;

namespace MyBooks.Service.Database
{
    public partial class KorisnikZnacka
    {
        public int Id { get; set; }
        public int KorisnikId { get; set; }
        public int ZnackaId { get; set; }
        public DateTime? DatumOtkljucavanja { get; set; }

        public virtual Korisnik Korisnik { get; set; } = null!;
        public virtual Znacka Znacka { get; set; } = null!;
    }
}

File: mybooks_mobile/lib/models/korisnik_znacka.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mybooks_mobile/models/znacka.dart';

part 'korisnik_znacka.g.dart';

@JsonSerializable()
class KorisnikZnacka {
  int id;
  int korisnikId;
  int znackaId;
  DateTime? datumOtkljucavanja;
  Znacka? znacka;

  KorisnikZnacka(this.id, this.korisnikId, this.znackaId,
      this.datumOtkljucavanja, this.znacka);

  factory KorisnikZnacka.fromJson(Map<String, dynamic> json) =>
      _$KorisnikZnackaFromJson(json);

  Map<String, dynamic> toJson() => _$KorisnikZnackaToJson(this);
}
